proposals/0000-ilp-ping-protocol/0000-ilp-ping-protocol.md

```diff
@@ -0,0 +1,100 @@
+# Interledger Ping Protocol
+
+## Introduction
+In order to monitor the health of an internetworked system, it is often useful to be able to send probe messages. In the Internet system, ICMP [echo](https://en.wikipedia.org/wiki/Ping_(networking_utility)#Echo_request) request and response packets are used together to verify connectivity (colloquially known as a "pinging" a host).  
+
+This RFC specifies a similar mechanism for Interledger nodes to test `uptime`, `response-time` and approximate `path cost` for a particular Initiator/Recipient pair.
+
+## Scope
+It is already possible to test end-to-end connectivity in Interledger by sending a small payment using existing mechanisms. However, doing so requires higher-level protocols, such as [SPSP](https://github.com/interledger/rfcs/blob/master/0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md) and [STREAM](https://github.com/interledger/rfcs/blob/master/0029-stream/0029-stream.md).
+
+Instead, this protocol uses only [Interledger-layer](https://github.com/interledger/rfcs/blob/master/0001-interledger-architecture/0001-interledger-architecture.md#interledger-protocol) technologies to test connectivity between two nodes in the Interledger without requiring any additional protocols or outside information.
+ 
+This protocol is not designed to debug routing issues and does not provide additional diagnostic information about the state of routing. That use case would be better served by a separate `traceroute` mechanism.
+
+All Interledger implementations _SHOULD_ respond to ping protocol requests, unless a node has a specific reason not to, such as security or privacy concerns.
```

Comment:
can we clarify implementation here? Is a STREAM server expected to respond to pings, or just connectors?

Reply:
hmmm...great question. I'm not sure I have a strong opinion. In Java-land, we've moved most of our receiver logic into the Connector runtime, so this deployment style would sort of get ping functionality for free.

I suppose after re-reading line 15 above, it's sort of optional for a STREAM receiver to support this functionality -- maybe that's enough?